src/ProductBundle/Entity/ProductCategoryManager.php

```diff
@@ -101,10 +101,9 @@ public function getProductCount(CategoryInterface $category, $limit = 1000)
                 AND (c.enabled = :categoryEnabled OR c.enabled IS NULL)
                 AND p.parent_id IS NULL
                 AND pc.category_id = :categoryId
-                LIMIT %d
```

Comment:
Have you seen the comment on line 87? Also, what are the consequences of removing this `LIMIT` for other RDBMSs?

Reply:
I update RP description.

Reply:
IMO, we should try to leverage the DBAL as much as possible in order to avoid dealing with specific RDBMSs by our own responsibility. This way, we could build the subquery using DQL or a query builder, and then if it is really required, get the raw SQL query to be used with the main query.
In the worst case, if that is not possible, we should build the different syntaxes by hand.
Even if this query is just intended to return the counted results, I guess leaving the limit open in the persistence layer is not good.

Reply:
@phansys have you seen the comment on line 87? @wbloszyk what about you?

Reply:
> @phansys have you seen the comment on line 87? @wbloszyk what about you?

Yes. What I mean is something like this:
```
$dql = 'SELECT DISTINCT(pc.product_id) FROM %s pc ... LIMIT %d';
$sql = 'SELECT count(cnt.product_id) AS "cntId" FROM('.$dql->getSQL().');';
```

Reply:
Oh right, I read to fast it seems… sorry!

Reply:
> > @phansys have you seen the comment on line 87? @wbloszyk what about you?
> 
> Yes. What I mean is something like this:
> 
> ```
> $dql = 'SELECT DISTINCT(pc.product_id) FROM %s pc ... LIMIT %d';
> $sql = 'SELECT count(cnt.product_id) AS "cntId" FROM('.$dql->getSQL().');';
> ```

- Subqueris is not allowed.
- Generating SQL crash (DQL look the same)
- Modify SQL for each platform is to mest

`ProductCategoryManager::getProductCount()` is only use in `ProductMenuBuilder::getCategoryTitle()`  to display counts in menu.

We can inject ProductManager and use:
https://github.com/sonata-project/ecommerce/blob/9c2a75ce27e2070eeb75349f2d542e49a0368756/src/ProductBundle/Entity/ProductManager.php#L65

This change will also limit products count when filters will be set.

Reply:
AFAIK, the syntax `SELECT COUNT(DISTINCT pc.product_id)` is [supported by DQL](https://www.doctrine-project.org/projects/doctrine-orm/en/2.5/changelog/migration_2_5.html#dql-improvements), even by query builder with `Expr::countDistinct()`. Why we need a subquery then?

Reply:
I don't know doctrine to much. Anyway we should get the same result from `ProductCategoryManager::getProductCount()` and `ProductManager::getProductCountInCategory` 
Names are example. We should reuse query as much as possible. It is also yours words. I will add some core example in Monday.

Reply:
Great! :+1:
Just let me know if you need help. I'm not using this bundle, but I can provide support with the ORM if required.